src/main/java/me/xmrvizzy/skyblocker/skyblock/item/exotic/ExoticCheck.java

```diff
@@ -0,0 +1,88 @@
+package me.xmrvizzy.skyblocker.skyblock.item.exotic;
+
+import me.xmrvizzy.skyblocker.skyblock.item.PriceInfoTooltip;
+import me.xmrvizzy.skyblocker.utils.Constants;
+import net.minecraft.nbt.NbtCompound;
+import net.minecraft.text.MutableText;
+import net.minecraft.text.Text;
+import net.minecraft.util.Formatting;
+
+public class ExoticCheck {
+    public static String getExpectedHex(String id) {
+        String color = PriceInfoTooltip.colorJson.get(id).getAsString();
```

Comment:
This causes the game to crash when the id isn't present, the result of `get` is null and because of that you cannot invoke `getAsString` and doing so causes the game to crash (happened to me when I was testing with a generic pair of fairy dyed leather boots)

I would do it like this which allows your null check to actually work: 
```java
String colorData = PriceInfoTooltip.colorJson.get(id);
String color = colorData != null ? colorData.getAsString() : null;
```